.buildkite/scripts/notify.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# This script creates a Buildkite notification step and pings the user who triggered the build.
+# It assumes the existence of the GitHub user's email.
+# If the email doesn't exist, the script attempts to extract it from the build message.
+# If no email is found, the default option will be used.
+
+set -euo pipefail
+
+# Define the default user not found message
+user_not_found="user_not_found"
+
+# Check if BUILDKITE_BUILD_CREATOR_EMAIL is set, defaulting to user not found
+build_creator="${BUILDKITE_BUILD_CREATOR_EMAIL:-$user_not_found}"
+build_message="${BUILDKITE_MESSAGE:-}"
+slack_channel="${SLACK_CHANNEL:-"#cloud-sec-ci"}"
+
+default_group_id=$(vault kv get -field="cloudbeat-eng-team" secret/ci/elastic-cloudbeat/slack-users)
+
+slack_user_id=$(vault kv get -field="$build_creator" secret/ci/elastic-cloudbeat/slack-users 2>/dev/null || echo "$user_not_found")
```

Comment:
nit: using "user_not_found" and then replacing it with `cloudbeat-eng-team` later is redundant, you can set `cloudbeat-eng-team` initially

Reply:
The script has been refactored